File: laocai-bot-spring-boot-starter/src/main/java/indi/kyson/laocai/bot/model/entity/GroupEntity.java
package indi.kyson.laocai.bot.model.entity;

/**
 * 群信息快照。
 * <p>
 * 事件处理和展示层只关心群的标识与容量信息，不需要直接依赖协议返回对象。
 * @param groupId 群号
 * @param groupName 群名称
 * @param memberCount 群成员数量
 * @param maxMemberCount 群容量
 */
public record GroupEntity(
        Long groupId,
        String groupName,
        Integer memberCount,
        Integer maxMemberCount
) { }

File: laocai-bot-spring-boot-starter/src/main/kotlin/indi/kyson/laocai/bot/entity/GroupEntity.kt
package indi.kyson.laocai.bot.entity

import com.fasterxml.jackson.annotation.JsonProperty

data class GroupEntity(
    @JsonProperty("group_id")
    val groupId: Long,
    @JsonProperty("group_name")
    val groupName: String,
    @JsonProperty("member_count")
    val memberCount: Int,
    @JsonProperty("max_member_count")
    val maxMemberCount: Int,
)
